### PR TITLE
Feat: DM image getter

### PR DESCRIPTION
**Pull request by alkihis**
Made small tweaks to allow image download from `ton.twitter.com`, which needs OAuth 1.0a signature to download DM images

Implement a new getter, `.downloadDmImage()`, available on `.v1` (with DM rights).

### CODE REVIEW

File: src/client-mixins/request-handler.helper.ts [hunk 1]
```diff
@@ -155,25 +155,43 @@ export class RequestHandlerHelper<T> {
...
+    }
+
+    return 'text';
+  }
+
+  protected getParsedResponse(res: IncomingMessage) {
+    const data = this.responseData;
+    const mode = this.requestData.forceParseMode || this.detectResponseType(res);
+
+    if (mode === 'buffer') {
```
Comment (PLhery) [inline src/client-mixins/request-handler.helper.ts:175]: [totally optional/nitpicking]: what about a switch-case?

File: src/v1/client.v1.ts [hunk 2]
```diff
@@ -242,6 +243,31 @@ export class TwitterApiv1 extends TwitterApiv1ReadWrite {
...
+  /**
+   * Get a single image attached to a direct message. TwitterApi client must be logged with OAuth 1.0a.
+   * https://developer.twitter.com/en/docs/twitter-api/v1/direct-messages/message-attachments/guides/retrieving-media
+   */
+  public async downloadDmImage(urlOrDm: string | DirectMessageCreateV1) {
+    if (typeof urlOrDm !== 'string') {
+      const attachment = urlOrDm[EDirectMessageEventTypeV1.Create].message_data.attachment;
+
+      if (!attachment) {
+        throw new Error('Given direct message doesnt contain attachment');
```
Comment (PLhery) [inline src/v1/client.v1.ts:258]: Nitpick/optional
```suggestion
        throw new Error('The given direct message doesn\'t contain any attachment');
```

**Review by PLhery — APPROVED** [review]

File: src/client-mixins/request-handler.helper.ts (hunk 1 above)
Comment (alkihis) [reply src/client-mixins/request-handler.helper.ts:175]: I don't like switches, especially when the condition block contains more than 1 instruction (case for json and url). That's why, but I can convert this code to `switch` if you want to.

Comment (PLhery) [reply src/client-mixins/request-handler.helper.ts:175]: no need to, thx for the explanation :)

**Pull request merged**